src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -405,6 +406,42 @@ private void FindThisPtrModification()
             }
         }
 
+        private Instantiation InstantiateGenericInstantiation(Instantiation instantiation)
+        {
+            var parameters = new TypeDesc[instantiation.Length];
+            for (int i = 0; i < instantiation.Length; i++)
+            {
+                if (instantiation[i].IsGenericParameter)
+                    parameters[i] = new InstantiatedGenericParameter((GenericParameterDesc)instantiation[i]);
+                else
+                    parameters[i] = instantiation[i];
+            }
+            return new Instantiation(parameters);
+        }
+
+        private void SetGenericInstantiationContext()
+        {
+            foreach (var inst in _method.OwningType.Instantiation)
+            {
+                if (inst is InstantiatedGenericParameter)
+                {
+                    var instantiatedGeneric = ((InstantiatedGenericParameter)inst);
```

Comment:
Can this be done in the InstantiatedGenericParameter constructor instead?

We try to avoid mutating state of already constructed objects.

Reply:
I am also not particularly happy with this...

Unfortunately for a more complex constraint to work, such as:
```
void GenericMethod<T, S>() where T : S where S : IEquatable<T>
```
the InstantiatedGeneric T needs to know of the InstantiatedGeneric S, which is only known after the InstantiatedGeneric T was created.

It might make sense to implement a method similar to the TypeSystemContext's `GetInstantiatedType` for InstantiatedGenerics and make the InstantiatedGeneric get all necessary other InstantiatedGenerics in its constructor this way.